### ISSUE TITLE
cwru数据集划分

### ISSUE THREAD

**Issue by miscedence12** [Feaxure-fresh/TL-Bearing-Fault-Diagnosis#27]
请问CWRU数据集应该如何划分，采样频率有12K和48K的，Normal样本好像也是在48K下采集的，请问我应该如何组织数据集？

**Comment by miscedence12** [comment]
@Feaxure-fresh

**Comment by miscedence12** [comment]
03-21 08:48:01 Source CWRU_1 detected 10 classes: ['Normal' 'ball_07' 'ball_14' 'ball_21' 'inner_07' 'inner_14' 'inner_21'
 'outer_07' 'outer_14' 'outer_21']
03-21 08:48:01 Target CWRU_5 detected 4 classes: ['Normal' 'ball_21' 'inner_21' 'outer_07']
03-21 08:48:01 The scenario is: partial domain adaptation
03-21 08:48:01 Using device cuda for training
Error.  nthreads must be a positive integer03-21 08:48:03 Source set CWRU_1 number of samples: 2014.
Label 0 has samples: 472
Label 1 has samples: 118
Label 2 has samples: 119
Label 3 has samples: 118
Label 4 has samples: 119
Label 5 has samples: 118
Label 6 has samples: 118
Label 7 has samples: 356
Label 8 has samples: 119
Label 9 has samples: 357
03-21 08:48:03 Training set number of samples: 1064.
Label 0 has samples: 472
Label 3 has samples: 118
Label 6 has samples: 118
Label 7 has samples: 356
03-21 08:48:03 Validation set number of samples: 1064.
Label 0 has samples: 472
Label 3 has samples: 118
Label 6 has samples: 118
Label 7 has samples: 356
03-21 08:48:04 -----Epoch 1/50-----
03-21 08:48:04 current lr: [0.01, 0.01]
100% 31/31 [00:05<00:00,  5.62it/s]
03-21 08:48:10 Train-Loss Source Classifier: 1.4986
03-21 08:48:10 Train-Loss Discriminator: 0.7036
03-21 08:48:10 Train-Acc Source Data: 49.09%
03-21 08:48:10 Train-Acc Discriminator: 50.00%
100% 17/17 [00:00<00:00, 74.10it/s]
03-21 08:48:10 Val-acc: 0.0620
03-21 08:48:10 The best model epoch 1, val-acc 6.20%
03-21 08:48:10 -----Epoch 2/50-----
03-21 08:48:10 current lr: [0.009990232305719944, 0.009990232305719944]
100% 31/31 [00:05<00:00,  5.92it/s]
03-21 08:48:16 Train-Loss Source Classifier: 0.4109
03-21 08:48:16 Train-Loss Discriminator: 0.6955
03-21 08:48:16 Train-Acc Source Data: 84.88%
03-21 08:48:16 Train-Acc Discriminator: 51.66%
100% 17/17 [00:00<00:00, 81.14it/s]
03-21 08:48:16 Val-acc: 0.8872
03-21 08:48:16 The best model epoch 2, val-acc 88.72%
03-21 08:48:16 -----Epoch 3/50-----
03-21 08:48:16 current lr: [0.009960967771506665, 0.009960967771506665]
100% 31/31 [00:05<00:00,  5.86it/s]
03-21 08:48:22 Train-Loss Source Classifier: 0.1379
03-21 08:48:22 Train-Loss Discriminator: 0.6740
03-21 08:48:22 Train-Acc Source Data: 94.46%
03-21 08:48:22 Train-Acc Discriminator: 66.61%
100% 17/17 [00:00<00:00, 100.92it/s]
03-21 08:48:22 Val-acc: 0.9868
03-21 08:48:22 The best model epoch 3, val-acc 98.68%
03-21 08:48:22 -----Epoch 4/50-----
03-21 08:48:22 current lr: [0.009912321891107008, 0.009912321891107008]
100% 31/31 [00:05<00:00,  5.93it/s]
03-21 08:48:27 Train-Loss Source Classifier: 0.0510
03-21 08:48:27 Train-Loss Discriminator: 0.6437
03-21 08:48:27 Train-Acc Source Data: 98.24%
03-21 08:48:27 Train-Acc Discriminator: 68.50%
100% 17/17 [00:00<00:00, 80.87it/s]
03-21 08:48:28 Val-acc: 0.9925
03-21 08:48:28 The best model epoch 4, val-acc 99.25%
03-21 08:48:28 -----Epoch 5/50-----
03-21 08:48:28 current lr: [0.009844486647586722, 0.009844486647586722]
100% 31/31 [00:05<00:00,  5.72it/s]
03-21 08:48:33 Train-Loss Source Classifier: 0.0238
03-21 08:48:33 Train-Loss Discriminator: 0.6177
03-21 08:48:33 Train-Acc Source Data: 99.45%
03-21 08:48:33 Train-Acc Discriminator: 70.44%
100% 17/17 [00:00<00:00, 98.01it/s]
03-21 08:48:33 Val-acc: 1.0000
03-21 08:48:33 The best model epoch 5, val-acc 100.00%  在使用DANN进行部分集任务训练时，DANN方法经常在几个epoch内的准确度达到了100%，这是否有问题？以上是训练日志

**Comment by Feaxure-fresh** [comment]
CWRU数据集划分是不是可以都用48k的与normal对齐？我觉得这个你靠实验决定吧，用12k的效果不好就只用48k的，这个也没有统一标准。我的实验细节太久远了我也忘了，反正框架是支持你自己组织数据集测试的。

至于你的效果100%我感觉不一定是有什么问题，CWRU迁移本身效果就比较好，更何况你选的源域和目标域一致性很高。不过你可以检查一下你的目标域是不是包含在源域里了，因为训练集和验证集样本量完全一致：
03-21 08:48:03 Training set number of samples: 1064.
Label 0 has samples: 472
Label 3 has samples: 118
Label 6 has samples: 118
Label 7 has samples: 356
03-21 08:48:03 Validation set number of samples: 1064.
Label 0 has samples: 472
Label 3 has samples: 118
Label 6 has samples: 118
Label 7 has samples: 356
况且你的样本量太少了，估计确实可以达到100%。

**Comment by miscedence12** [comment]
是的，我是把目标域的全部样本都拿来参与训练过程，因为目标域样本是无标签的，然后测试的时候也是用的全部目标域样本

**Comment by miscedence12** [comment]
这里的TrainingSet和Validation se应该都是取自于目标域样本，你的代码里应该是这样的@Feaxure-fresh

**Comment by Feaxure-fresh** [comment]
我的意思是你可以检查一下源域和目标域是不是有样本重合，因为你的源域每个label的样本数与目标域完全一致，是不是可能你目标域已经是源域的一部分了，实际上模型已经从源域上得到了目标域的标签，所以准确率很高。就是你的CWRU_1和CWRU_5是不是完全不同的工况做迁移。如果没问题的话可能就是准确率很高，或许是CWRU_1和CWRU_5相似度很高，评估样本量又少，所以迁移效果很高。

**Comment by miscedence12** [comment]
@Feaxure-fresh 好的谢谢解答
